Kernel/Storage/ATA/AHCIPort.cpp

```diff
@@ -22,7 +22,8 @@ namespace Kernel {
 
 UNMAP_AFTER_INIT NonnullRefPtr<AHCIPort> AHCIPort::must_create(AHCIController const& controller, AHCI::HBADefinedCapabilities hba_capabilities, volatile AHCI::PortRegisters& registers, u32 port_index)
 {
-    auto port = adopt_ref_if_nonnull(new (nothrow) AHCIPort(controller, hba_capabilities, registers, port_index)).release_nonnull();
+    auto identify_buffer_page = MUST(MM.allocate_supervisor_physical_page());
+    auto port = adopt_ref_if_nonnull(new (nothrow) AHCIPort(controller, *identify_buffer_page, hba_capabilities, registers, port_index)).release_nonnull();
```

Comment:
Why are we transferring this as a C++ reference that is then turned into a string RefPtr reference inside? That seems pretty rickety. Can't we just `move()` it directly?